### PR TITLE
[SITE] Guest post for fail2ban setup

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: src/homepage/_posts/2023-06-20-fail2ban.markdown [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+title:  "Guest post: James and fail2ban"
+date:   2023-05-17  01:16:30 +0200
+categories: james update
+---
+
+Credits: Paul Gunter
+
+[Original][Original] version.
+
+# Repel brute force attackes with Linux firewall (iptables) and fail2ban
```
Comment (Arsnael) [inline src/homepage/_posts/2023-06-20-fail2ban.markdown:12]:
```suggestion
# Repel brute force attacks with Linux firewall (iptables) and fail2ban
```

File: src/homepage/_posts/2023-06-20-fail2ban.markdown [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+Fail2ban is set up with the following commands:
+
+```
+supo apt update
+sudo apt install fail2ban
+sudo systemctl start fail2ban
+sudo systemctl enable fail2ban
+```
+
+Ubuntu installs the software. After start fail3ban manually. Finally the service is set to start
```
Comment (Arsnael) [inline src/homepage/_posts/2023-06-20-fail2ban.markdown:129]:
```suggestion
Ubuntu installs the software. After start fail2ban manually. Finally the service is set to start
```

File: src/homepage/_posts/2023-06-20-fail2ban.markdown [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+configuration file for this.
+
+We build a layout that outputs the following information:
+
+ - The timestamp (%d)
+ - The Java class from which the entry is generated (%C)
+ - The attacker's IP address (%X)
+ - The log message (%msg)
+ - A line break (%n)
+ -
```
Comment (Arsnael) [inline src/homepage/_posts/2023-06-20-fail2ban.markdown:222]:
```suggestion

```

**Review by Arsnael — APPROVED** [review]
Few small typos but looks great to me, very instructive :)

Thanks to Paul Gunter for writing this up!

**Review by quantranhong1999 — APPROVED** [review]
Interesting article.

**Review by Arsnael — APPROVED** [review]

**Review by vttranlina — APPROVED** [review]

File: src/homepage/_posts/2023-06-20-fail2ban.markdown [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+
+```
+# "bantime" is the number of seconds that a host is banned.
+bantime = 10m
+# A host is banned if it has generated "maxretry" during the last "findtime"
+# seconds.
+findtime = 10m
+# "maxretry" is the number of failures before a host get banned.
+maxretry = 5
+In our case we change the values only for the SSHD service in the „/etc/fail2ban/jail.d/defaultsdebian.conf“ file. After the file looks like this:
```
Comment (guenterpaul) [inline src/homepage/_posts/2023-06-20-fail2ban.markdown:188]: This is Text, co code

File: src/homepage/_posts/2023-06-20-fail2ban.markdown [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+
+```
+[Definition]
+failregex =
+^.-.org.apache.james.protocols.smtp.core.fastfail.AbstractValidRcptHandler.\
+[ip=<HOST>\].-.Rejected message. Unknown user.*$
+ ^.-.org.apache.james.protocols.smtp.core.esmtp.AuthCmdHandler.\[ip=\].-.AUTH
+method LOGIN failed from Username.*\@<HOST>$
+ignoreregex =
+We see 2 regular expressions there. The first looks at the „AbstractValidRcptHandler“ class. The IP
```
Comment (guenterpaul) [inline src/homepage/_posts/2023-06-20-fail2ban.markdown:315]: This and the next 4 lines are text, no code

**Review by guenterpaul — CHANGES_REQUESTED** [review]
some lines are text - not code

File: src/homepage/_posts/2023-06-20-fail2ban.markdown [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+```
+# "bantime" is the number of seconds that a host is banned.
+bantime = 10m
+# A host is banned if it has generated "maxretry" during the last "findtime"
+# seconds.
+findtime = 10m
+# "maxretry" is the number of failures before a host get banned.
+maxretry = 5
+In our case we change the values only for the SSHD service in the „/etc/fail2ban/jail.d/defaultsdebian.conf“ file. After the file looks like this:
+[sshd]
```
Comment (chibenwa) [inline src/homepage/_posts/2023-06-20-fail2ban.markdown:189]:
```suggestion
maxretry = 5
```

In our case we change the values only for the SSHD service in the „/etc/fail2ban/jail.d/defaultsdebian.conf“ file. After the file looks like this:

```
[sshd]
```

File: src/homepage/_posts/2023-06-20-fail2ban.markdown [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+[ip=<HOST>\].-.Rejected message. Unknown user.*$
+ ^.-.org.apache.james.protocols.smtp.core.esmtp.AuthCmdHandler.\[ip=\].-.AUTH
+method LOGIN failed from Username.*\@<HOST>$
+ignoreregex =
+We see 2 regular expressions there. The first looks at the „AbstractValidRcptHandler“ class. The IP
+address is filled via the James variable. The IP address is recognized with the placeholder <HOST>.
+The second looks at the "AuthCmdHandler" class, the IP address is at the end of the line.
+Setup Jail
+We create the file "/etc/fail2ban/jail.d/james.conf" with the following content:
+[james]
```
Comment (chibenwa) [inline src/homepage/_posts/2023-06-20-fail2ban.markdown:320]:
````suggestion
ignoreregex =
```
We see 2 regular expressions there. The first looks at the „AbstractValidRcptHandler“ class. The IP
address is filled via the James variable. The IP address is recognized with the placeholder <HOST>.

The second looks at the "AuthCmdHandler" class, the IP address is at the end of the line.

## Setup Jail

We create the file "/etc/fail2ban/jail.d/james.conf" with the following content:
```
[james]
````

**Comment by guenterpaul** [comment]
@chibenwa Thanks a lot for changes. There are still some problems with line breaks and spaces. I think my PDF file was the problem. The files are "log4j2.xml", "james.log" and "james.conf". There are the lines:  
[code.txt](https://github.com/apache/james-project/files/11829586/code.txt)

**Pull request merged**